frontend/src/screens/GameScreen/gameScreen.tsx

```diff
@@ -0,0 +1,247 @@
+        setTurkishWords(newTurkishWords)
+      }
+    }
+
+    const removeWordFunction = (turkishWords: MaybeWordPair[], word: MaybeWordPair, i: number) => {
+      return () => {
+        const newTurkishWords = [...turkishWords]
+        newTurkishWords[i] = { turkish: word.turkish, correctEnglishWord: word.correctEnglishWord }
+        setTurkishWords(newTurkishWords)
+    }
+    }
+
+    const renderTurkishWords = () => {
+        return turkishWords?.map((word, i) => (
+            <DroppableRow
+                key={word.turkish}
+                showingResults={submitted}
+                wordDropped={wordDroppedFunction(turkishWords, word, i)}
+                turkish={word.turkish}
+                english={word.english}
+                correctEnglish={word.correctEnglishWord}
+                removeWord={removeWordFunction(turkishWords, word, i)}
+                isPairing={props.isPairing}
+            />))
+    }
+
+    const renderSubmittedButtons = () => {
+        return (<View style={styles.doneContainer}><PrimaryButton
+            disabled={isLoading}
+            onPress={() => {
+                setIsLoading(true);
+                restartRound()
+            }}
+            text="play again" />
+            <SecondaryButton
+                disabled={isLoading}
+                text="end session"
+                onPress={() => {
+                    fi.endRound(currentRoundId);
+                    fi.endSession(sessionId);
+                    navigation.navigate("HomeScreen");
+                }} />
+        </View>)
+    }
+
+    const renderInProgressButtons = () => {
+        return (<View style={styles.doneContainer}>
+            <PrimaryButton
+                disabled={!canClickDoneButton || isLoading}
+                text="done"
+                onPress={() => setSubmitted(true)} />
+        </View>
+        )
+    }
+
+    const correctValues = turkishWords?.filter(({ english, correctEnglishWord }) => english === correctEnglishWord).length ?? 0;
+    const canClickDoneButton = englishWords.every((word) => turkishWords?.some(({ english }) => english === word))
+
+    const scoreText = props.topScreenRenderFunction(correctValues, turkishWords?.length ?? 0)
+    const shouldFlexWrap = props.isPairing && !submitted
+
+    return (
+        <DraxProvider>
+            <View style={styles.container}>
+                <View style={shouldFlexWrap ? styles.wrapTopContainer : styles.noWrapTopContainer }>
+                    {submitted ? scoreText : renderEnglishOptions() }
+                </View>
+                <View style={styles.bottomContainer}>
+                    {renderTurkishWords()}
+                </View>
+                {submitted ? renderSubmittedButtons() : renderInProgressButtons()}
+            </View>
+        </DraxProvider >
+    )
+}
+
+
+const defaultStyle = StyleSheet.create({
```

Comment:
This is only used by `styles.english` so could we remove it and just have these in `styles.english`?